.github/workflows/test-appetize.yml

```diff
@@ -5,12 +5,37 @@ on:
   pull_request:
     types: [labeled, synchronize]
     branches:
+      - deployment/*
       - master
 jobs:
+  # Set the deployment name, to be used as the name of the environment in which to run the following job
+  # Pattern found in this question: https://github.com/actions/runner/issues/998#issue-817330769
+  set_deployment_name:
+    if: contains(github.event.pull_request.labels.*.name, 'Test - appetize')
+    outputs:
+      DEPLOYMENT_NAME: ${{ steps.get_deployment_name.outputs.DEPLOYMENT_NAME }}
+    runs-on: ubuntu-latest
+    steps:
+    # Extract deployment name as the slug of target PR branch via string operators, e.g. /deployment/plh_tz -> plh_tz
+    # https://www.linuxjournal.com/article/8919, https://github.com/greenbone/gsa/actions/runs/1277581365/workflow
+    - name: Get deployment name
+      id: get_deployment_name
+      run: |
+        if [ "$GITHUB_BASE_REF" = "master" ];                    \
+        then echo "DEPLOYMENT_NAME=plh_global" >> $GITHUB_OUTPUT;              \
+        else echo "DEPLOYMENT_NAME=${GITHUB_BASE_REF##*/}" >> $GITHUB_OUTPUT;  \
+        fi;
   test_appetize:
+    needs: set_deployment_name
     if: contains(github.event.pull_request.labels.*.name, 'Test - appetize')
     runs-on: ubuntu-latest
+    environment: 
+      name: ${{ needs.set_deployment_name.outputs.DEPLOYMENT_NAME }}
+    env:
+      DEPLOYMENT_NAME: ${{ needs.set_deployment_name.outputs.DEPLOYMENT_NAME }}
     steps:
+      - name: Log Deployment/Environment Name
+        run: echo $DEPLOYMENT_NAME
```

Comment:
nit(non-blocking)
I think when going through action logs it shows the environment variables anyway so no need to log beforehand, e.g.
![image](https://user-images.githubusercontent.com/10515065/228228296-a40fa63f-24b5-49e1-8893-3ce42a58af47.png)

Reply:
Ah yeh I didn't spot that. I was thinking it could be helpful to log what environment is being used, but I note that that's also available elsewhere, at least on the PR and the specific Github Actions page
<img width="500" alt="Screenshot 2023-03-28 at 13 43 34" src="https://user-images.githubusercontent.com/64838927/228239866-8e851ec9-81f9-43bf-bdf4-0ceeaa31c69c.png">
<img width="600" alt="Screenshot 2023-03-28 at 13 43 10" src="https://user-images.githubusercontent.com/64838927/228239870-8688b07b-48ee-4550-80ef-cabaa31594cb.png">